File: mod16/main.ts
function countUnvisitedStones(stones: number, birds: number[]): number {
    const visited: boolean[] = new Array(stones + 1).fill(false);
    for (const bird of birds) {
        for (let i = bird; i <= stones; i += bird) {
            visited[i] = true;
        }
    }
    let unvisitedCount = 0;
    for (let i = 1; i <= stones; i++) {
        if (!visited[i]) {
            unvisitedCount++;
        }
    }
    return unvisitedCount;
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Введите количество камней: ', (stonesInput: string) => {
    const stones = parseInt(stonesInput);
    readline.question('Введите расстояния, которые могут преодолеть птицы (через пробел): ', (birdsInput: string) => {
        const birds = birdsInput.split(' ').map(bird => parseInt(bird));
        console.log(`Количество непосещенных камней: ${countUnvisitedStones(stones, birds)}`);
        readline.close();
    });
});

File: mod16/main.rs
use std::io;

fn count_unvisited_stones(stones: i32, birds: &[i32]) -> i32 {
    let mut visited = vec![false; (stones + 1) as usize];
    for &bird in birds {
        let mut i = bird;
        while i <= stones {
            visited[i as usize] = true;
            i += bird;
        }
    }
    (1..=stones).filter(|&i| !visited[i as usize]).count() as i32
}

fn main() {
    let mut stones_input = String::new();
    println!("Введите количество камней:");
    io::stdin().read_line(&mut stones_input).expect("Не удалось прочитать строку");
    let stones: i32 = stones_input.trim().parse().expect("Некорректный ввод");

    let mut birds_input = String::new();
    println!("Введите расстояния, которые могут преодолеть птицы (через пробел):");
    io::stdin().read_line(&mut birds_input).expect("Не удалось прочитать строку");
    let birds: Vec<i32> = birds_input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Некорректный ввод"))
        .collect();

    println!("Количество непосещенных камней: {}", count_unvisited_stones(stones, &birds));
}